src/worker/dbWorker.ts

```diff
@@ -13,58 +14,24 @@ interface WorkerMessage {
     payload: GSetPayload;
 }
 
-let gDB: IDBDatabase | null = null; // Global reference to the IndexedDB instance
+const dbUtil = IndexedDBUtil.getInstance();  // Use the singleton instance of IndexedDBUtil
 const sets: Map<string, GSet<any>> = new Map(); // Map to track GSet instances
 
-// Function to open a connection to the IndexedDB
-function openDB(): Promise<IDBDatabase> {
-    return new Promise((resolve, reject) => {
-        const request = indexedDB.open('SynchronizationDB', 1);
-        request.onupgradeneeded = (event) => {
-            const db = (event.target as IDBOpenDBRequest).result;
-            if (!db.objectStoreNames.contains('crdts')) {
-                db.createObjectStore('crdts', { keyPath: 'id' });
-            }
-        };
-        request.onsuccess = () => {
-            resolve(request.result);
-        };
-        request.onerror = () => reject(request.error);
-    });
-}
-
-// Function to store a GSet in IndexedDB
-function storeSet(db: IDBDatabase, id: string, gSet: GSet<any>): Promise<void> {
-    return new Promise((resolve, reject) => {
-        const transaction = db.transaction(['crdts'], 'readwrite');
-        const store = transaction.objectStore('crdts');
-        const request = store.put({ id, items: gSet.getItems() });
-
-        transaction.onerror = (event) => {
-            // More robust error handling for the transaction
-            reject((event.target as IDBRequest).error);
-        };
-
-        request.onsuccess = () => resolve();
-        request.onerror = () => reject(request.error);
-    });
-}
-
 // Function to add an item to a GSet and store it in IndexedDB
-function addToSet(db: IDBDatabase, id: string, item: any): Promise<void> {
+async function addToSet(id: string, item: any): Promise<void> {
     let gSet = sets.get(id) || new GSet<any>();
     gSet.add(item);
     sets.set(id, gSet);
-    return storeSet(db, id, gSet);
+    await dbUtil.addSet(id, gSet);

```

Comment:
The `await` is redundant

```suggestion
    dbUtil.addSet(id, gSet);
```